service-netty/src/main/java/info/bitrich/xchangestream/service/netty/NettyStreamingService.java

```diff
@@ -226,6 +222,8 @@ protected void handleError(CompletableEmitter completable, Throwable t) {
                 isManualDisconnect = false;
             }
         });
+        // shutdown sockets after disconnect for avoiding sockets leak
+        eventLoopGroup.shutdownGracefully(2, 30, TimeUnit.SECONDS);
```

Comment:
I wonder if there might be any potential race conditions from the fact that this is an asynchronous call. `completable.onError(t)` will fire immediately and the shutdown will occur in the background, possibly while a retry reconnect starts.

I am not sure.  Anyone else?

Reply:
I agree it fixes the leak, for sure.

I think I'm convinced on the async too. Good! Thanks for the explanations.